backend/hct_mis_api/apps/payment/migrations/0140_migration.py

```diff
@@ -0,0 +1,233 @@
+        fsp.delivery_mechanisms.set(fsp_dms)
+
+
+def data_migration_deliverymechanismdata_delivery_mechanism(apps, schema_editor):
+    DeliveryMechanismData = apps.get_model("payment", "DeliveryMechanismData")
+    DeliveryMechanism = apps.get_model("payment", "DeliveryMechanism")
+    dms = list(DeliveryMechanism.objects.all())
+
+    dmds = DeliveryMechanismData.objects.exclude(delivery_mechanism_choice__isnull=True)
+    for dmd in dmds:
+        dmd_dm = [dm for dm in dms if dm.name == dmd.delivery_mechanism_choice][0]
+        dmd.delivery_mechanism = dmd_dm
+
+    DeliveryMechanismData.objects.bulk_update(dmds, ["delivery_mechanism"])
+
+
+def data_migration_deliverymechanismperpaymentplan_delivery_mechanism(apps, schema_editor):
+    DeliveryMechanismPerPaymentPlan = apps.get_model("payment", "DeliveryMechanismPerPaymentPlan")
+    DeliveryMechanism = apps.get_model("payment", "DeliveryMechanism")
+    dms = list(DeliveryMechanism.objects.all())
+
+    dmppps = DeliveryMechanismPerPaymentPlan.objects.exclude(delivery_mechanism_choice__isnull=True)
+    for dmppp in dmppps:
+        dmppp_dm = [dm for dm in dms if dm.name == dmppp.delivery_mechanism_choice][0]
+        dmppp.delivery_mechanism = dmppp_dm
+
+    DeliveryMechanismPerPaymentPlan.objects.bulk_update(dmppps, ["delivery_mechanism"])
+
+
+def data_migration_fspxlsxtemplateperdeliverymechanism_delivery_mechanism(apps, schema_editor):
+    FspXlsxTemplatePerDeliveryMechanism = apps.get_model("payment", "FspXlsxTemplatePerDeliveryMechanism")
+    DeliveryMechanism = apps.get_model("payment", "DeliveryMechanism")
+    dms = list(DeliveryMechanism.objects.all())
+
+    fspxtdms = FspXlsxTemplatePerDeliveryMechanism.objects.exclude(delivery_mechanism_choice__isnull=True)
+    for fspxtdm in fspxtdms:
+        fspxtdm_dm = [dm for dm in dms if dm.name == fspxtdm.delivery_mechanism_choice][0]
+        fspxtdm.delivery_mechanism = fspxtdm_dm
+
+    FspXlsxTemplatePerDeliveryMechanism.objects.bulk_update(fspxtdms, ["delivery_mechanism"])
+
+
+def data_migration_payment_delivery_type(apps, schema_editor):
+    Payment = apps.get_model("payment", "Payment")
+    DeliveryMechanism = apps.get_model("payment", "DeliveryMechanism")
+    dms = list(DeliveryMechanism.objects.all())
+
+    payments = Payment.objects.exclude(delivery_type_choice__isnull=True)
+    for payment in payments:
+        payment_dm = [dm for dm in dms if dm.name == payment.delivery_type_choice][0]
+        payment.delivery_type = payment_dm
+
+    Payment.objects.bulk_update(payments, ["delivery_type"])
+
+
+def data_migration_payment_record_delivery_type(apps, schema_editor):
+    PaymentRecord = apps.get_model("payment", "PaymentRecord")
+    DeliveryMechanism = apps.get_model("payment", "DeliveryMechanism")
+    dms = list(DeliveryMechanism.objects.all())
+
+    payment_records = PaymentRecord.objects.exclude(delivery_type_choice__isnull=True)
+    for payment_record in payment_records:
```

Comment:
Make for Delivery mechanism
then 
Payment.filter(delivery_mechanism='CONSTANT').update(set_deliverymechanism_fk)